subsys/bluetooth/host/scan.c

```diff
@@ -712,32 +713,36 @@ void bt_hci_le_per_adv_sync_established(struct net_buf *buf)
 	}
 
 	if (!pending_per_adv_sync ||
-	    (!atomic_test_bit(pending_per_adv_sync->flags,
-			      BT_PER_ADV_SYNC_SYNCING_USE_LIST) &&
+	    (!atomic_test_bit(pending_per_adv_sync->flags, BT_PER_ADV_SYNC_SYNCING_USE_LIST) &&
 	     ((pending_per_adv_sync->sid != evt->sid) ||
 	      bt_addr_le_cmp(&pending_per_adv_sync->addr, &evt->adv_addr)))) {
-		struct bt_le_per_adv_sync_term_info term_info;
-
 		BT_ERR("Unexpected per adv sync established event");
+		/* Request terminate of pending periodic advertising in controller */
 		per_adv_sync_terminate(sys_le16_to_cpu(evt->handle));
 
+		unexpected_evt = true;
+	} else {
+		unexpected_evt = false;
+	}
+
+	if ((unexpected_evt == false && evt->status != BT_HCI_ERR_SUCCESS) ||
+	    unexpected_evt == true) {
 		if (pending_per_adv_sync) {
+			struct bt_le_per_adv_sync_term_info term_info;
+
 			/* Terminate the pending PA sync and notify app */
 			term_info.addr = &pending_per_adv_sync->addr;
 			term_info.sid = pending_per_adv_sync->sid;
+			term_info.err = evt->status;
```

Comment:
So the reason why `bt_le_per_adv_sync_term_info` didn't have a `err` field is that the `bt_hci_evt_le_per_adv_sync_lost` doesn't. 

We can add it no problem, as it definitely serves a purpose, but we should also initialize the value to something in `bt_hci_le_per_adv_sync_lost` as well.

Reply:
Added missing initialization in `bt_hci_le_per_adv_sync_lost`.